### ISSUE TITLE
Class not available for a basket plugin

### ISSUE THREAD

**Issue by verschuur** [aimeos/aimeos-core#123]
Hi, I'm trying to create a basket plugin but I keep getting the following error on the front end of the site: 

`Class "\Aimeos\MShop\Plugin\Provider\Order\ApproveOrder" not available`

This is on version 2017.10

**Steps:**
1. Created my package _np-approve_order_ using the Aimeos extension builder.
2. Followed the instructions at [Implement basket plugins](https://aimeos.org/docs/Developers/Library/Implement_basket_plugins)
3. Used the skeleton class from above link to create my plugin, which is now at 
`/ext/np-approve-order/lib/custom/src/MShop/Plugin/Provider/Order/ApproveOrder.php`
4. The class namespace is `Aimeos\MShop\Plugin\Provider\Order` (as per the skeleton class) and the classname is `ApproveOrder`.
5. The manifest.php contains the `lib/custom/src` folder in the includes array.

So far, so good. However, when I go the admin plugin page and I try to add the plugin, it isn't availble in the list of provider. The class also doesnt seem to be loaded by composer, even after `composer dump-autoload` and `aimeos:cache`.

I think I am missing a step or two somewhere, possible a service provider or something along those lines. Looking through the documentation for [creating an extension](https://aimeos.org/docs/Developers/Create_an_extension), I don't see further instructions on how to register your extension.

I've tried looking at the other packages in the extension folder, but I couldnt find an answer there. 

So basically, how do I register an extension and/or basket plugin?

**Comment by aimeos** [comment]
Can you post the code of your class you've created?
Normally, you don't need any additional registration as long as your `./lib/custom/src` path is in the `includes` section of the `manifest.php` of your extension. Please also make sure the extension you've created is for 2017.x, not for the current 2018.x version

**Comment by verschuur** [comment]
I fixed it with the help of Norbert :)

It turned out I was working in the `/ext` folder which, of course, isn't loaded by the autoload. I added the path `ext/np-approve-order/lib/custom/src` to the Composer classmap and everything worked as expected.

Thanks again Norbert!
